File: easy/intersection-of-two-arrays-ii.py
def checkIntersection(moreNums, lessNums):
  result = []
  for num in lessNums:
    if num in moreNums:
      result.append(num)
      moreNums.remove(num)
  return result

def intersect(nums1, nums2):
  if len(nums1) > len(nums2):
    return checkIntersection(nums1, nums2)
  else:
    return checkIntersection(nums2, nums1)

print(intersect([4,9,5], [9,4,9,8,4]))

File: easy/intersection-of-two-arrays-ii.js
var intersect = function(nums1, nums2) {
  if(nums1.length > nums2.length) {
      return checkIntersection(nums1, nums2)
  } else {
      return checkIntersection(nums2, nums1)
  }
};

const checkIntersection = (moreNums, lessNums) => {
  const result = []
  for(const num of lessNums) {
      if(moreNums.includes(num)) {
          result.push(num)
          moreNums.splice(moreNums.indexOf(num), 1)
      }
  }
  return result
}

console.log(intersect([1,2,2,1], [2,2]))